x-pack/legacy/plugins/snapshot_restore/public/app/components/policy_form/steps/step_settings.tsx

```diff
@@ -0,0 +1,450 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { Fragment, useState } from 'react';
+
+import {
+  EuiDescribedFormGroup,
+  EuiTitle,
+  EuiFormRow,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiSpacer,
+  EuiSwitch,
+  EuiLink,
+  EuiSelectable,
+  EuiPanel,
+  EuiComboBox,
+} from '@elastic/eui';
+import { Option } from '@elastic/eui/src/components/selectable/types';
+import { SlmPolicyPayload, SnapshotConfig } from '../../../../../common/types';
+import { documentationLinksService } from '../../../services/documentation';
+import { useAppDependencies } from '../../../index';
+import { StepProps } from './';
+
+export const PolicyStepSettings: React.FunctionComponent<StepProps> = ({
+  policy,
+  indices,
+  updatePolicy,
+  errors,
+}) => {
+  const {
+    core: { i18n },
+  } = useAppDependencies();
+  const { FormattedMessage } = i18n;
+  const { config = {} } = policy;
+
+  const updatePolicyConfig = (updatedFields: Partial<SlmPolicyPayload['config']>): void => {
+    const newConfig = { ...config, ...updatedFields };
+    updatePolicy({
+      config: newConfig,
+    });
+  };
+
+  // States for choosing all indices, or a subset, including caching previously chosen subset list
+  const [isAllIndices, setIsAllIndices] = useState<boolean>(!Boolean(config.indices));
+  const [cachedIndices, setCachedIndices] = useState<SnapshotConfig['indices']>([...indices]);
```

Comment:
I am not sure I understand why this is called a cache. It seems to be a indices user "selection". 

```js
const [indicesSelection, setIndicesSelection] = ... 
```